tests/Controller/ProductReviewApiTest.php

```diff
@@ -0,0 +1,359 @@
+        'HTTP_Authorization' => 'Bearer SampleTokenNjZkNjY2MDEwMTAzMDkxMGE0OTlhYzU3NzYyMTE0ZGQ3ODcyMDAwM2EwMDZjNDI5NDlhMDdlMQ',
+        'ACCEPT' => 'application/json',
+    ];
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_show_product_review_list_when_access_is_denied()
+    {
+        $productReviewsData = $this->loadFixturesFromFile('resources/product_reviews.yml');
+
+        /** @var ProductInterface $product */
+        $product = $productReviewsData['product1'];
+
+        $this->client->request('GET', $this->getReviewListUrl($product));
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'authentication/access_denied_response', Response::HTTP_UNAUTHORIZED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_show_product_review_when_it_does_not_exist()
+    {
+        $productReviewsData = $this->loadFixturesFromFile('resources/product_reviews.yml');
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        /** @var ProductInterface $product */
+        $product = $productReviewsData['product1'];
+
+        $this->client->request('GET', $this->getReviewListUrl($product) . '0', [], [], static::$authorizedHeaderWithAccept);
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_showing_product_review()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $productReviewsData = $this->loadFixturesFromFile('resources/product_reviews.yml');
+
+        /** @var ProductInterface $product */
+        $product = $productReviewsData['product1'];
+
+        /** @var ReviewInterface $productReview */
+        $productReview = $productReviewsData['productReview1'];
+
+        $this->client->request('GET', $this->getReviewUrl($product, $productReview), [], [], static::$authorizedHeaderWithAccept);
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'product_review/show_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_indexing_product_reviews()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $productReviewsData = $this->loadFixturesFromFile('resources/product_reviews.yml');
+
+        /** @var ProductInterface $product */
+        $product = $productReviewsData['product1'];
+
+        $this->client->request('GET', $this->getReviewListUrl($product), [], [], static::$authorizedHeaderWithAccept);
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'product_review/index_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_create_product_review()
```

Comment:
`it_allows_creating_new_product_review`

Reply:
Hello @Zales0123,

For these methods names I checked also tests/Controller/ProductVariantApiTest.php witch have added the same method name's logic: "it_allows_create_product_variant"; it means that for  tests/Controller/ProductVariantApiTest.php is also incorect? if not, what is the difference between ProductReviewApiTest.php and ProductVariantApiTest.php, why for one is correct and for the other one, not?

Thank you.
Paul.

Reply:
For the name of the methods the most important requirement is language correctness :) So even if somewhere else method names are named incorrectly (from the grammatical point of view) we should on writing new ones as perfect as possible.

Reply:
Alright, then I will rewrite, these methods name, with the good ones, for tests/Controller/ProductReviewApiTest.php.